### PR TITLE
Unused opens analyzer

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
image

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/CodeFix/RemoveUnusedOpens.fs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace rec Microsoft.VisualStudio.FSharp.Editor
```
Comment (cartermp) [inline vsintegration/src/FSharp.Editor/CodeFix/RemoveUnusedOpens.fs]: Why `namespace rec` here?

Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.Editor/CodeFix/RemoveUnusedOpens.fs]: Removed.

File: src/fsharp/vs/ServiceAssemblyContent.fs [hunk 2]
```diff
@@ -24,8 +24,8 @@ module internal Extensions =
     [<RequireQualifiedAccess>]
     [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
     module Option =
-        let inline attempt (f: unit -> 'T) = try Some (f()) with _ -> None        
-        let inline orElse v = function Some x -> Some x | None -> v
+        let attempt (f: unit -> 'T) = try Some (f()) with _ -> None        
+        let orElse v = function Some x -> Some x | None -> v
```
Comment (vasily-kirichenko) [inline src/fsharp/vs/ServiceAssemblyContent.fs]: Done.

**Comment by vasily-kirichenko** [comment]
It works pretty well, no false positives on whole FSharp.Editor project.

**Comment by cartermp** [comment]
Found issue:

```fsharp
module M =
    module N =
        let (>>>>) x y = x + y

open M.N // Says I can remove this.

[<EntryPoint>]
let main argv = 
    let foo = 12 >>>> 13
    0
```

remove-unused

**Comment by KevinRansom** [comment]
@vasily-kirichenko , @cartermp 

I'm going to pull this as is, can you raise an issue for resolving the issue.
Thanks

Kevin

**Comment by vasily-kirichenko** [comment]
@KevinRansom thanks, Kevin

**Comment by dsyme** [comment]
@KevinRansom I'm taking a look at the perf for this feature, but it seems like the whole feature was checked in without any tests – either for FCS or for vsintegration. This can't happen in the future please.

**Comment by vasily-kirichenko** [comment]
I’m pretty sure it won’t happen again ;)

**Comment by vasily-kirichenko** [comment]
@dsyme actually there are a small number of tests https://github.com/Microsoft/visualfsharp/blob/master/vsintegration/tests/unittests/UnusedOpensTests.fs

**Comment by dsyme** [comment]
@vasily-kirichenko Thanks, I has only searched under tests/service, my bad.

**Pull request merged**